src/Symfony/Component/Validator/Constraints/VideoValidator.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Kev <https://github.com/symfonyaml>
+ */
+class VideoValidator extends FileValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Video) {
+            throw new UnexpectedTypeException($constraint, Video::class);
+        }
+
+        if (!class_exists(\FFMpeg\FFProbe::class)) {
```

Comment:
no need for this check: if a Video instance is given, it's been already verified

Reply:
@nicolas-grekas the `Video` class could be extended and the constructor skipped.